en/computer_vision/collision_prevention.md

```diff
@@ -57,13 +57,14 @@ If so, the vehicle velocity is restricted.
 This velocity restriction takes into account both the inner velocity loop tuned by [MPC_XY_P](../advanced_config/parameter_reference.md#MPC_XY_P), as well as the [jerk-optimal velocity controller](../config_mc/mc_jerk_limited_type_trajectory.md) via [MPC_JERK_MAX](../advanced_config/parameter_reference.md#MPC_JERK_MAX) and [MPC_ACC_HOR](../advanced_config/parameter_reference.md#MPC_ACC_HOR).
 The velocity is restricted such that the vehicle will stop in time to maintain the distance specified in [CP_DIST](#CP_DIST).
 The range of the sensors for each sector is also taken into account, limiting the velocity via the same mechanism.
+the restricted velocity is then transformed into an acceleration with [MPC_XY_VEL_P_ACC](../advanced_config/parameter_reference.md#MPC_XY_VEL_P_ACC).
```

Comment:
FWIW as above, I don't really understand what this means. I think that perhaps we're always controlling the acceleration setpoint, so for our controller, so what this says is that we've worked out what our restricted velocity is supposed to be, and this converts it to the value that we need for our accel. setpoint?

Reply:
```suggestion
The restricted velocity is then transformed into an acceleration with [MPC_XY_VEL_P_ACC](../advanced_config/parameter_reference.md#MPC_XY_VEL_P_ACC).
```

Reply:
Yea, its a bit complicated. which is also why the rewrite has not been done before.
we have our velocity and acceleration setpoint coming from the operator. and the problem is that they can point into different directions. so we need to compensate the current velocity if its heading into an obstacle, and restrict the acceleration setpoint from the operator.
the current approach worked fine with velocities. which is why i take the constrained velocity with the old approach, subtract the current velocity to get a velocity error, and  just add the proportional gain of the acceleration controller (the input to whom is an velocity error). you can kind of think of the gain as something which modifies the units, input is [m/s] output is [m/s²]. (in an abstract sense)

and for the setpoint of the operator, i just scale it in regard of the distance to the minimum distance. the closer the vehicle is to the minimum distance, the more the normal component gets scaled to zero.